solution/backend/file_manager/models.py

```diff
@@ -24,6 +24,7 @@ class Meta:
     full_name = models.CharField(max_length=512, null=False, blank=False)
     short_name = models.CharField(max_length=50, null=True, blank=True)
     abbreviation = models.CharField(max_length=10, null=True, blank=True)
+    combined_sort = CombinedNaturalSort(['short_name', 'abbreviation', 'full_name'], null=True)
```

Comment:
Sticks with the current hierarchy of ordering in shortname, abbreviation, and full name.  We do this on the front end.